src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Security/DoNotDisableTokenValidationChecksTests.cs

```diff
@@ -0,0 +1,230 @@
+        parameters.ValidateTokenReplay = flag;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestLiteralNoDiagnostic()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+using System;
+using Microsoft.IdentityModel.Tokens;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        var parameters = new TokenValidationParameters();
+        parameters.RequireExpirationTime = true;
+        parameters.ValidateAudience = true;
+        parameters.ValidateIssuer = true;
+        parameters.ValidateLifetime = true;
+        parameters.RequireAudience = true;
+        parameters.RequireSignedTokens = true;
+        parameters.ValidateActor = true;
+        parameters.ValidateIssuerSigningKey = true;
+        parameters.ValidateTokenReplay = true;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestConstantNoDiagnostic()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+using System;
+using Microsoft.IdentityModel.Tokens;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        const bool flag = true;
+        var parameters = new TokenValidationParameters();
+        parameters.RequireExpirationTime = flag;
+        parameters.ValidateAudience = flag;
+        parameters.ValidateIssuer = flag;
+        parameters.ValidateLifetime = flag;
+        parameters.RequireAudience = flag;
+        parameters.RequireSignedTokens = flag;
+        parameters.ValidateActor = flag;
+        parameters.ValidateIssuerSigningKey = flag;
+        parameters.ValidateTokenReplay = flag;
+    }
+}");
+        }
+
+        [Fact]
+        public async Task TestPropertyInitializerNoDiagnostic()
+        {
+            await VerifyCSharpAnalyzerAsync(@"
+using System;
+using Microsoft.IdentityModel.Tokens;
+
+class TestClass
+{
+    public void TestMethod()
+    {
+        var parameters = new TokenValidationParameters
+        {
+            RequireExpirationTime = true,
+            ValidateAudience = true,
+            ValidateIssuer = true,
+            ValidateLifetime = true,
+            RequireAudience = true,
+            RequireSignedTokens = true,
+            ValidateActor = true,
+            ValidateIssuerSigningKey = true,
+            ValidateTokenReplay = true
+        };
+    }
+}");
+        }
+
+        private static async Task VerifyCSharpAnalyzerAsync(string source, params DiagnosticResult[] expected)
+        {
+            var csharpTest = new VerifyCS.Test
+            {
+                ReferenceAssemblies = AdditionalMetadataReferences.DefaultWithWilson,
+                TestState =
+                {
+                    Sources = { source },
+                },
+            };
+
+            csharpTest.ExpectedDiagnostics.AddRange(expected);
+
+            await csharpTest.RunAsync();
+        }
+
+        private static DiagnosticResult GetCSharpResultAt(int line, int column)
+            => VerifyCS.Diagnostic()
+                .WithLocation(line, column);
```

Comment:
@Evangelink Didn't you ban this overload before? 😨 

@TimHannMSFT Please remove this helper method and use the markup syntax instead. This works by surrounding the code where the diagnostic is expected between `[|` and `|]`

Example:

https://github.com/dotnet/roslyn-analyzers/blob/762b08948cdcc1d94352fba681296be7bf474dd7/src/NetAnalyzers/UnitTests/Microsoft.NetCore.Analyzers/Runtime/ForwardCancellationTokenToInvocationsTests.cs#L478

Reply:
I can certainly have a look, seems this is used in quite a few of the other test classes though:

DoNotDisableCertificateValidationTests.cs, DoNotDisableHTTPHeaderCheckingTests.cs, etc

Reply:
@TimHannMSFT Yes it's still used in some tests. But I believe it's a goal to get rid of them as time goes on. That's why such overloads were banned in https://github.com/dotnet/roslyn-analyzers/pull/4631. I'm not sure why there is no errors here for using the banned symbols.

Reply:
`WithLocation()` isn't banned in the `2.9.x` branch (guess there wasn't an altnernative?). I'd prefer sticking with the old style in both `2.9.x` and `main` to keep merges simpler when there are fixes. We can add `#pragma warning disable RS0030` in `main` like we do for other security analyzer tests.

Reply:
Sounds good, I went through and did this before seeing Paul's comment but I stashed those changes and will leave as is for now.